x/gov/keeper/tally.go

```diff
@@ -35,30 +35,31 @@ func (keeper Keeper) Tally(ctx sdk.Context, proposal types.Proposal) (passes boo
 
 	keeper.IterateVotes(ctx, proposal.ProposalID, func(vote types.Vote) bool {
 		// if validator, just record it in the map
-		// if delegator tally voting power
-		valAddrStr := sdk.ValAddress(vote.Voter).String()
-		if val, ok := currValidators[valAddrStr]; ok {
+		voteValAddrStr := sdk.ValAddress(vote.Voter).String()
+		if val, ok := currValidators[voteValAddrStr]; ok {
 			val.Vote = vote.Option
-			currValidators[valAddrStr] = val
-		} else {
-			// iterate over all delegations from voter, deduct from any delegated-to validators
-			keeper.sk.IterateDelegations(ctx, vote.Voter, func(index int64, delegation exported.DelegationI) (stop bool) {
-				valAddrStr := delegation.GetValidatorAddr().String()
+			currValidators[voteValAddrStr] = val
+		}
 
-				if val, ok := currValidators[valAddrStr]; ok {
-					val.DelegatorDeductions = val.DelegatorDeductions.Add(delegation.GetShares())
-					currValidators[valAddrStr] = val
+		// iterate over all delegations from voter, deduct from any delegated-to validators
+		keeper.sk.IterateDelegations(ctx, vote.Voter, func(index int64, delegation exported.DelegationI) (stop bool) {
+			valAddrStr := delegation.GetValidatorAddr().String()
 
-					delegatorShare := delegation.GetShares().Quo(val.DelegatorShares)
-					votingPower := delegatorShare.MulInt(val.BondedTokens)
+			val, ok := currValidators[valAddrStr]
+			// there is no need to deduct amounts from their own validator
```

Comment:
We can probably just deduct from self-validator for simplicity reasons.  No need to keep this as a separate case.  It should work the same way.

Reply:
if this is okay - we should still include a comment discussing why this is okay

Reply:
bump @Ruihuan :)

Reply:
Should I revert the deduction judgement to original or just keep this code and then add some comment?

Reply:
It seems we should remove the special case and add a comment explaining it.